ethsigner/src/main/java/tech/pegasys/ethsigner/http/JsonRpcErrorHandler.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.http;
+
+import static io.netty.handler.codec.rtsp.RtspResponseStatuses.GATEWAY_TIMEOUT;
+import static io.netty.handler.codec.rtsp.RtspResponseStatuses.INTERNAL_SERVER_ERROR;
+import static tech.pegasys.ethsigner.jsonrpc.response.JsonRpcError.DOWNSTREAM_NODE_TIMED_OUT;
+import static tech.pegasys.ethsigner.jsonrpc.response.JsonRpcError.INTERNAL_ERROR;
+
+import tech.pegasys.ethsigner.jsonrpc.JsonRpcRequest;
+import tech.pegasys.ethsigner.jsonrpc.JsonRpcRequestId;
+import tech.pegasys.ethsigner.jsonrpc.exception.JsonRpcException;
+import tech.pegasys.ethsigner.jsonrpc.response.JsonRpcError;
+import tech.pegasys.ethsigner.jsonrpc.response.JsonRpcErrorResponse;
+
+import java.util.Optional;
+
+import io.vertx.core.Handler;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.Json;
+import io.vertx.ext.web.RoutingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JsonRpcErrorHandler implements Handler<RoutingContext> {
+  private static final Logger LOG = LoggerFactory.getLogger(LogErrorHandler.class);
+  private HttpResponseFactory httpResponseFactory;
+
+  public JsonRpcErrorHandler(HttpResponseFactory httpResponseFactory) {
+    this.httpResponseFactory = httpResponseFactory;
+  }
+
+  @Override
+  public void handle(final RoutingContext context) {
+    final Optional<JsonRpcRequest> jsonRpcRequest = jsonRpcRequest(context);
+    final JsonRpcErrorResponse errorResponse = errorResponse(context, jsonRpcRequest);
+    final int statusCode =
+        context.statusCode() == -1 ? INTERNAL_SERVER_ERROR.code() : context.statusCode();
+    final String body = jsonRpcRequest.map(Json::encodePrettily).orElse(context.getBodyAsString());
```

Comment:
As this body reference is only used in the debug log statement, move it into a lambda for lazy execution _i.e. only when LogLevel.DEBUG is used

Reply:
done